website/content/en/docs/scorecard/custom-tests.md

```diff
@@ -0,0 +1,185 @@
+
+```yaml
+- name: "check-version"
+  image: quay.io/$USERNAME/scorecard-test:dev
+  entrypoint: 
+  - scorecard-test
+  - check-version
+  labels:
+    suite: generic
+    test: check-version-test
+  description: check if csv has version
+```
+
+The important fields to note here are:
+1. `image` - name and tag of the test image which was specified in the Makefile.
+2. `labels` - the name of the `test` and `suite` the test function belongs to. This can be specified in the `operator-sdk alpha scorecard` command to run the desired test.
+
+### Include test definition in Scorecard binary:
+
+Modify [main.go][scorecard_main_func] file under `operator-sdk/images/scorecard-test/cmd/test/` to include the test label specifier.
+
+```Go
+...     
+switch entrypoint[0] {
+case tests.OLMBundleValidationTest:
+	result = tests.BundleValidationTest(tmpDir)
+case tests.OLMCRDsHaveValidationTest:
+	result = tests.CRDsHaveValidationTest(*cfg)
+case tests.OLMCRDsHaveResourcesTest:
+	result = tests.CRDsHaveResourcesTest(*cfg)
+case tests.OLMSpecDescriptorsTest:
+	result = tests.SpecDescriptorsTest(*cfg)
+case tests.OLMStatusDescriptorsTest:
+	result = tests.StatusDescriptorsTest(*cfg)
+case tests.BasicCheckSpecTest:
+    result = tests.CheckSpecTest(*cfg)
+case tests.CheckVersionTest:
+    result = tests.CheckVersionTest(*cfg)
+default:
+	result = printValidTests()
+}
+...
+```
+
+Include custom test to the list of valid tests in the `printValidTests()` function:
+
+```Go
+func printValidTests() (result v1alpha2.ScorecardTestResult) {
+...
+    str := fmt.Sprintf("Valid tests for this image include: %s, %s, %s, %s, %s, %s",
+            tests.OLMBundleValidationTest,
+            tests.OLMCRDsHaveValidationTest,
+            tests.OLMCRDsHaveResourcesTest,
+            tests.OLMSpecDescriptorsTest,
+            tests.OLMStatusDescriptorsTest,
+            tests.BasicCheckSpecTest,
+            tests.CheckVersionTest)
+    result.Errors = append(result.Errors, str)
+...
+
+```
+
+### Run scorecard command
+
+Run the following command to execute the scorecard tests. 
+
+```console
+operator-sdk alpha scorecard --config=internal/scorecard/alpha/testdata/bundle/tests/scorecard/config.yaml --selector=suite=generic -o json --wait-time=32s --bundle=internal/scorecard/alpha/testdata/bundle/ --skip-cleanup=false
+```
+
+The name or suite of the tests which are to be executed can be specified with the `--selector` flag. The command will create scorecard pods with the image specified in `config.yaml` for the respective test. For example, the `CheckVersionTest` test provides the following json output for a CSV with version specified.
+
+```console
+{
+  "metadata": {
+    "creationTimestamp": null
+  },
+  "log": "",
+  "results": [
+    {
+      "name": "check-version",
+      "description": "test if this custom function works",
+      "state": "pass",
+      "log": "Loaded ClusterServiceVersion: memcached-operator.v0.0.1\n"
+    }
+  ]
+}
```

Comment:
If we change the tool output it will be outdated easily. So, IMO would be better to add an example of the commands that should be executed in each step with the description then add their outputs.

Reply:
This is the output of `operator-sdk scorecard --alpha` command. I have modified it to be included with the command.